Contributing.md

```diff
@@ -178,6 +178,7 @@ Summary:
 - If a compiler warning is found, please try and fix it.
 - Try to avoid using raw pointers (pointers allocated with `new`) as much as possible. There are cases where using a raw pointer is unavoidable, and in these situations it is OK to use them. An example of this is functions from a C library that require them. In cases where it is avoidable, the STL usually has a means to solve this (`vector`, `unique_ptr`, etc).
 - Do not use the `auto` keyword everywhere. While it's nice that the type can be determined by the compiler, it cannot be resolved at 'readtime' by the developer as easily. Use auto only in cases where it is obvious what the type being assigned is (note: 'obvious' means not having to open other files or reading the header file). Some situations where it is appropriate to use `auto` is when iterating over a `std::map` container in a foreach loop, or to shorten the length of container iterator variable declarations.
+- Retain the asterix when using `auto` to refer to pointer types (`auto* =` instead of `auto =`).
```

Comment:
`asterisk`. This should likely also mention the ampersands for references.

Reply:
Not exactly related - from the point of view of the compiler, `auto x = ` is the same as `auto* x =` when the RHS is a pointer, so this is actually a style issue.

When the RHS is not a pointer, `auto x = ` is different from `auto& x =`  
The former is a copy in that case which you may actually want.

Reply:
Right, that's true; I overlooked that. Never mind about mentioning ampersands then.